File: September-2025/01_09_2025_gfg.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    int sumOfModes(vector<int>& arr, int k) {
        int n = arr.size();
        int sum = 0;
        for (int i = 0; i <= n - k; ++i) {
            vector<int> subarray;
            for (int j = 0; j < k; ++j) {
                subarray.push_back(arr[i + j]);
            }

            unordered_map<int, int> counts;
            int max_freq = 0;
            int mode = -1;

            for (int num : subarray) {
                counts[num]++;
                if (counts[num] > max_freq) {
                    max_freq = counts[num];
                    mode = num;
                } else if (counts[num] == max_freq && num < mode) {
                    mode = num;
                }
            }
            sum += mode;
        }
        return sum;
    }
};

File: September-2025/01_09_2025_gfg.py
def sum_of_modes(arr, k):
    """
    Calculates the sum of the modes of all subarrays of size k in the given array.

    Args:
        arr: A list of positive integers.
        k: An integer representing the size of the subarrays.

    Returns:
        The sum of the modes of all subarrays of size k.
    """
    n = len(arr)
    modes_sum = 0
    for i in range(n - k + 1):
        subarray = arr[i:i + k]
        counts = {}
        for num in subarray:
            counts[num] = counts.get(num, 0) + 1
        
        max_freq = 0
        mode = float('inf')
        for num, freq in counts.items():
            if freq > max_freq:
                max_freq = freq
                mode = num
            elif freq == max_freq and num < mode:
                mode = num
        modes_sum += mode
    return modes_sum